src/main/java/com/teragrep/jos_01/procfs/Task.java

```diff
@@ -0,0 +1,135 @@
+ * of the requirements of the GNU Affero GPL version 3 as long as this Program
+ * is the same Program as licensed from Suomen Kanuuna Oy without any additional
+ * modifications.
+ *
+ * Supplemented terms under GNU Affero General Public License version 3
+ * section 7
+ *
+ * Origin of the software must be attributed to Suomen Kanuuna Oy. Any modified
+ * versions must be marked as "Modified version of" The Program.
+ *
+ * Names of the licensors and authors may not be used for publicity purposes.
+ *
+ * No rights are granted for use of trade names, trademarks, or service marks
+ * which are in The Program if any.
+ *
+ * Licensee must indemnify licensors and authors for any liability that these
+ * contractual assumptions impose on licensors and authors.
+ *
+ * To the extent this program is licensed as part of the Commercial versions of
+ * Teragrep, the applicable Commercial License may apply to this file if you as
+ * a licensee so wish it.
+ */
+package com.teragrep.jos_01.procfs;
+
+import com.teragrep.jos_01.procfs.status.GenericStatus;
+import com.teragrep.jos_01.procfs.status.ProcessStat;
+
+import java.io.File;
+import java.util.ArrayList;
+
+public class Task {
+
+    private final long taskId;
+    private final long processId;
+    private final File procDirectory;
+    private ArrayList<GenericStatus> processStatuses;
+
+    public Task(String taskId, Process parentProcess) {
+        this(Long.parseLong(taskId), parentProcess);
+    }
+
+    public Task(long taskId, Process parentProcess) {
+        this.taskId = taskId;
+        this.processId = parentProcess.pid();
+        this.procDirectory = new File("/proc/" + processId + "/task/", Long.toString(processId));
+        this.processStatuses = new ArrayList<GenericStatus>();
+    }
+
+    // Creates a Status object based on the chosen file name in /proc for this process. Overloaded to accept different kinds of Status objects
+    private GenericStatus reportStatistics(String procFileName) {
+        ArrayList<String> rows = readProcFile(procFileName);
+        GenericStatus status = new GenericStatus(rows);
+        processStatuses.add(status);
+        return status;
+    }
+
+    private ArrayList<String> readProcFile(String procFileName) {
+        ProcFile procFile = new ProcFile(procDirectory, procFileName);
+        ArrayList<String> rows = procFile.readFile();
+        return rows;
+    }
+
+    public GenericStatus proc(String procFileName) {
+        ArrayList<String> rows = readProcFile(procFileName);
+        return new GenericStatus(rows);
+    }
+
+    public ProcessStat stat() {
+        ArrayList<String> rows = readProcFile("stat");
+        return new ProcessStat(rows);
+    }
+
+    public ArrayList<String> availableProcFiles() {
+        ArrayList<String> fileNames = procFileNames(new ArrayList<String>(), procDirectory);
+        return fileNames;
+    }
+
+    private ArrayList<String> procFileNames(ArrayList<String> nameList, File file) {
```

Comment:
I think we can just remove this method entirely, it was used in a previous prototype but that has since been refactored and this method no longer has any uses.